### PR TITLE
Edit / polish website content

### PR DESCRIPTION
**Pull request by fifthposition**
Revised the text on home page, "Clients", "Drivers", and "Resources". Some small formatting changes too.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -2,60 +2,65 @@
...
 * Connection URLs
-* Java ServiceLoader-based `ConnectionFactory` discover and configuration
+* `ConnectionFactory` discovery and configuration based on Java's `ServiceLoader`
 * Observability
 
 ## Latest Release
-The latest release of R2DBC is 0.8.0.RELEASE. You can [download the spec](/spec/0.8.0.RELEASE/spec/html/) or [browse the javadoc](/spec/0.8.0.RELEASE/api/).
 
-## [Driver Implementations](/drivers/)
+The latest release of R2DBC is 0.8.0.RELEASE. You can [download the spec](/spec/0.8.0.RELEASE/spec/html/) or [browse the Javadoc](/spec/0.8.0.RELEASE/api/).
```
Comment (gregturn) [inline index.md]: This should get bumped to 0.8.1.RELEASE.

File: z003-drivers.md [hunk 2]
```diff
@@ -6,53 +6,56 @@ permalink: /drivers/
...
 
-* Group: **io.r2dbc**
-* Artifact: **r2dbc-spi**
+* <strong>Group</strong> <code>io.r2dbc</code>
+* <strong>Artifact</strong> <code>r2dbc-spi</code>
 
-The key interface all driver providers must implement is the [`Connection`](https://r2dbc.io/spec/0.8.0.RELEASE/api/io/r2dbc/spi/Connection.html) and a set of other interfaces.
+* **Group** `io.r2dbc`
+* **Artifact** `r2dbc-spi`
+
```
Comment (gregturn) [inline z003-drivers.md]: May wish to be consistent on apply strong and code formatting.

File: z003-drivers.md [hunk 3]
```diff
@@ -6,53 +6,56 @@ permalink: /drivers/
...
+The key interface that all driver providers must implement is the [`Connection`](https://r2dbc.io/spec/0.8.0.RELEASE/api/io/r2dbc/spi/Connection.html), along with a set of other interfaces.
 Check out the specification for details on [R2DBC Driver Compliance](/spec/0.8.0.RELEASE/spec/html/#compliance).
```
Comment (gregturn) [inline z003-drivers.md]: Again, we're now at 0.8.1.RELEASE.

File: z003-drivers.md (hunk 2 above)
Comment (fifthposition) [reply z003-drivers.md]: Yeah, I think I just used this to describe the artifact. I might instead pull this out from being a list and put the artifact and group inline.

Comment (fifthposition) [reply z003-drivers.md]: Ah, I see what happened… I had left one version with HTML tags and another with Markdown, so there were two listings for the same thing. Oops. Fixed! :)

**Pull request merged**